test/browser/async_pthread_join.c

```diff
@@ -0,0 +1,30 @@
+// Copyright 2023 The Emscripten Authors.  All rights reserved.
```

Comment:
Can you put this new test under `test/pthread` and also add it to `test_core.py` so we can run it under node?

Reply:
Can you also give the file that same name as the test.. I find it makes files easier to find.  i.e. `test_pthread_join_and_asyncify.c`

Reply:
For now this has to be a browser test, since it needs pthreads and JSPI. When we get a newer version of node we could convert it.

Reply:
We do do testing with node canary these days.   See `require_wasm_eh` in test/common.py.     You would write a similar `require_jspi` and then add the JSPI tests to the list  to `test-node-compat` in .circleci/config.yml